src/sliding_sync.rs

```diff
@@ -132,46 +133,105 @@ async fn build_client(
     ))
 }
 
-
-async fn login(cli: Cli) -> Result<(Client, Option<String>)> {
-    let (client, client_session) = build_client(&cli, app_data_dir()).await?;
-
-    // Query the server for supported login types.
-    let login_kinds = client.matrix_auth().get_login_types().await?;
-    if !login_kinds.flows.iter().any(|flow| matches!(flow, LoginType::Password(_))) {
-        bail!("Homeserver does not support username + password login flow.");
+/// Logs in to the given Matrix homeserver using the given username and password.
+///
+/// This function is used by the login screen to log in to the Matrix server.
+///
+/// The function returns the client and an optional error message if the login fails.
+/// If the login succeeds, the function returns `None` for the error message.
+/// 
+/// The second returned value is sync token
+///
+/// The third returned value is a boolean indicating whether the login was successful.
+/// This value is used by the login screen to determine whether to show the login form again
+/// or to show the main app UI after a successful login.
```

Comment:
A function should never return an "optional" error message within the `Result::Ok()` variant, that's very strange. The good thing is that you're not actually doing that, and it looks like the comment is just out of date.

```suggestion
/// Upon success, this function returns the logged-in client and an optional sync token.
```